docs/scos/dev/the-docker-sdk/202204.0/the-docker-sdk.md

```diff
@@ -15,12 +15,219 @@ related:
 3. Simplifying the process of setting up a local demo of Spryker project.
+
+This document provides details on the Docker SDK itself, its commands, and repository structure.
+
+## Docker SDK repository structure
+
+The [docker-sdk repository](https://github.com/spryker/docker-sdk) contains the following folders:
+
+* [bin](https://github.com/spryker/docker-sdk/tree/master/bin): structured sh commands that are copied and used in the containers.
+* [ci](https://github.com/spryker/docker-sdk/tree/master/ci): Travis deployment pipeline, basic tests.
+* [context](https://github.com/spryker/docker-sdk/tree/master/context): service configurations, such as PHP, MySCL, nginx, etc.
+* [deployment](https://github.com/spryker/docker-sdk/tree/master/deployment): generated Docker images, configs, and assets that are necessary to build containers.
+* [docs](https://github.com/spryker/docker-sdk/tree/master/docs): documentation.
+* [generator](https://github.com/spryker/docker-sdk/tree/master/generator): generator of the `deployment` folder.
+* [images](https://github.com/spryker/docker-sdk/tree/master/images): Docker files for services, applications, etc.
+  
+## Docker/sdk boot command details
+
+The Docker SDK `boot` command runs the [generator](https://github.com/spryker/docker-sdk/tree/master/generator) application. The generator is deployed as a Docker container. The generator
+takes twig templates and project's deploy.yml file with context and generates content of the [deployment](https://github.com/spryker/docker-sdk/tree/master/deployment) folder.
+
+The `boot` command does the following:
+1. Prepares the `env` files for `glue`,`zed`, `yves`.
+2. Prepares endpoints and store-specific configs.
+3. Prepares the Nginx configuration for the frontend and gateway containers.
+4. Prepares the PHP and debug configuration.
+5. Transforms the `env` files.
+6. Prepares Dashboard.
+7. Prepares the Docker image files and the `docker-compose.yml` file for local development.
+8. Generates bash scripts in `deploy.sh`.
+9. Shows the list of commands that are necessary to complete installation.
+
+## Docker/sdk up command details
+
+The Docker SDK `up` command does the following:
+
+1. Prepares data in `deployment/default/deploy.sh`:
+   1. Mounts Mutagen: 
+      1. Removes the sync volume
+      2. Terminates Mutagen
+      3. Creates the volume
+   2. Builds app images in CLI with the `docker/sdk build` command.
+   3. Tags application images with the `docker/sdk tag` command.
+   4. Mounts codebase with the following commands:
+      - `install composer`
+      - `install -s build`
+      - `build-{mode}`
+   5. Mounts assets with the following commands:
+      - `install -s build-static`
+      - `build-static-{mode}`
+   6. Builds frontend and gateway containers from an Nginx image.
+2. Runs containers:
+   1. Runs before stack
+   2. Runs containers
+   3. Runs after stack
+3. Loads data for each region:
+   1. Skips if DB is not empty: if `data`, then `force`.
```

Comment:
Skips if DB is not empty. If `--data` argument is provided, then the command will be "forced"